### PR TITLE
LazyLoad and AutoPoll refresh logic improvements

### PR DESCRIPTION
**Pull request by adams85**
### Describe the purpose of your pull request

* Improves refresh logic of `AutoPollConfigService` and `LazyLoadConfigService` to take the cache download time into account to fetch the config only if cached config is unavailable or stale.
* Also does some cleanup around `*ConfigService` classes.

Breaking changes:
* Slightly changes the behavior of `ProjectConfig.Timestamp` (only updated when download succeeds, regardless of whether the config has changed or not). ~Note: **this may need further discussion** as it slightly differs from e.g. the .NET SDK behavior, which updates timestamp not only when receiving status code 200 or 304 but also on any successful communication with the CDN servers, regardless of the returned status code. In case we want to fully align the behavior across these SDKs, we need to change the implementations of `IConfigFetcher` in the JS SDKs because they don't provide the necessary information currently.~ This is addressed by https://github.com/configcat/common-js/pull/66

### Requirement checklist (only if applicable)

- [x] I have covered the applied changes with automated tests.
- [x] I have executed the full automated test set against my changes.
- [x] I have validated my changes against all supported platform versions.
- [x] I have read and accepted the [contribution agreement](https://github.com/configcat/legal/blob/main/contribution-agreement.md).

### CODE REVIEW

File: src/ProjectConfig.ts [hunk 1]
```diff
@@ -3,22 +3,21 @@ export class ProjectConfig {
...
 
     /**
      * Determines whether the specified ProjectConfig instances are considered equal.
      */
     public static equals(projectConfig1: ProjectConfig | null, projectConfig2: ProjectConfig | null): boolean {
-        if (!projectConfig1 || !projectConfig2) return false;
-
-        return this.compareEtags(projectConfig1.HttpETag, projectConfig2.HttpETag);
+        // If both configs are null, we consider them equal.
+        return projectConfig1 ? !!projectConfig2 && this.compareEtags(projectConfig1.HttpETag, projectConfig2.HttpETag) : !projectConfig2;
```
Comment (endret) [inline src/ProjectConfig.ts:20]: Please implement some cover tests when the configs are null/not null.

**Review by endret — APPROVED** [review]
